src/java/voldemort/store/readonly/ReadOnlyStorageEngine.java

```diff
@@ -176,6 +178,27 @@ public String getChunkIdToNumChunks() {
         return builder.toString();
     }
 
+    public void reSyncFileSystem() {
+        fileModificationLock.readLock().lock();
+        File latestSymLink = new File(latestDirSymLinkPath);
+        boolean needToResync = !Utils.symlinkExists(ReadOnlyUtils.getCurrentVersion(storeDir), latestSymLink);
```

Comment:
Are you going to leak the `fileModificationLock` if `symlinkExists()` throws an exception?